Typescript build fails when using `moduleResolution` `node16` or `nodenext`

### Question 💬

When switching `moduleResolution` in `tsconfig.json` to `node16` or `nodenext` the build fails.

### How to reproduce ☕️

```json
{
  "name": "test",
  "version": "1.0.0",
  "type": "module",
  "scripts": {
    "build": "tsc --build",
    "clean": "tsc --build --clean"
  },
  "dependencies": {
    "@auth/core": "^0.7.0"
  },
  "devDependencies": {
    "@types/nodemailer": "^6.4.7",
    "typescript": "^5.0.4"
  }
}
```

```ts
import type { AuthConfig } from '@auth/core'
import GithubProvider from '@auth/core/providers/github'

const config: AuthConfig = {
  secret: process.env.AUTH_SECRET,
  providers: [
    GithubProvider({
      clientId: process.env.GITHUB_CLIENT_ID!,
      clientSecret: process.env.GITHUB_CLIENT_SECRET!,
    }),
  ],
}
```

Build error:

```console
node_modules/.pnpm/@auth+core@0.7.0/node_modules/@auth/core/providers/credentials.d.ts:3:34 - error TS2307: Cannot find module 'preact/src/jsx.js' or its corresponding type declarations.

3 import type { JSXInternal } from "preact/src/jsx.js";
```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

--- a/packages/core/src/providers/credentials.ts
+++ b/packages/core/src/providers/credentials.ts
@@ -1,6 +1,6 @@
 import type { CommonProviderOptions } from "./index.js"
 import type { Awaitable, User } from "../types.js"
-import type { JSXInternal } from "preact/src/jsx.js"
+import type { JSX } from "preact"
 
 /**
  * Besides providing type safety inside {@link CredentialsConfig.authorize}
@@ -8,7 +8,7 @@ import type { JSXInternal } from "preact/src/jsx.js"
  * on the default sign in page.
  */
 export interface CredentialInput
-  extends Partial<JSXInternal.IntrinsicElements["input"]> {
+  extends Partial<JSX.IntrinsicElements["input"]> {
   label?: string
 }
 

